file/src/test/java/docs/javadsl/FileTailSourceTest.java

```diff
@@ -112,22 +118,56 @@ public void willReadNewLinesAppendedAfterReadingTheInitialContents() throws Exce
     subscriber.expectComplete();
   }
 
+  // #shutdown-on-delete
+
+  private static class Tick {}
+
+  /**
+   * Periodically checks if the file at `path` exists every `pollingInterval`. Shutdown the stream
+   * when the file is not found using an empty Source.
+   * @param pollingInterval Interval to check if file exists.
+   * @param path a file to check
+   */
+  private static Source<String, Cancellable> fileCheckSource(java.time.Duration pollingInterval, Path path) {
+    return Source
+            .tick(pollingInterval, pollingInterval, new Tick())
+            .mapConcat(tick -> {
+              if (Files.exists(path)) {
+                return Collections.<String>emptyList();
+              }
+              throw new FileNotFoundException();
+            })
+            .recoverWith(new PFBuilder<Throwable, Source<String, NotUsed>>()
+                    .match(FileNotFoundException.class, t -> Source.empty())
+                    .build()
+            );
+  }
```

Comment:
One thing that struck me: don't we have a FS-change-event-source-connector as well, couldn't one just filter-map that down to only deletes and this file?

Reply:
Yes, that would be a valid use case as well to manage the stream from the outside. The original purpose of this example was to shutdown the stream from within when a file was deleted. You could create a graph that uses the `DirectoryChangesSource` to do the same thing within, but it would be scanning for filesystem changes on a whole directory instead of individual files.

Reply:
Good point.  I'll take a look at that soon.

Reply:
I updated the example to use `DirectoryChangesSource`.  I noticed that it's a bit slower to detect the event then eagerly checking if the file exists, no matter what I set the interval frequenty to.  It required me to increase the test probe timeout to `akka.test.single-expect-default = 10s` so the test would pass.